src/framework/script/script-type.js

```diff
@@ -310,9 +34,27 @@ class ScriptType extends EventHandler {
         return this.__attributes;
     }
 
+    /**
+     * @protected
+     * @param {*} args - initialization arguments
+     */
+    initScript(args) {
+        // super does not exist due to the way the class is instantiated
+        Script.prototype.initScript.call(this, args);
+        this.__attributes = { };
+        this.__attributesRaw = args.attributes || { }; // need at least an empty object to make sure default attributes are initialized
+    }
+
+    /**
+     * Expose initScript as initScriptType for backwards compatibility
```

Comment:
Is this intended to be public API in the API ref manual?

Reply:
Nope, marking as @protected